cpp/oneapi/dal/detail/common.hpp

```diff
@@ -49,6 +49,28 @@
     } while (0)
 #endif
 
+#ifdef ONEDAL_DATA_PARALLEL
+#define __ONEDAL_IF_QUEUE__(optional_queue, ...) \
```

Comment:
How sound the rule to create a macro with underscores?

Reply:
This is to let users know, what is allowed to use. Macros with `__` are internal and should not be used. Macro w/o `__` is "safe" to use outside library.